### PR TITLE
Update old scripts examples

### PR DESCRIPTION
**Pull request by Olyno**
### Description
Update old examples of Skript.

---
**Target Minecraft Versions:**     /
**Requirements:**     /
**Related Issues:**     #2252

### CODE REVIEW

File: src/main/resources/scripts/vanilla gui.sk [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#
+# A vanilla example of a gui creation using Skript only
+# Requiert Skript 2.3+
```
Comment (JRoy) [inline src/main/resources/scripts/vanilla gui.sk]: How would the user have this script without skript 2.3 :P

File: src/main/resources/scripts/simple motd.sk [hunk 2]
```diff
@@ -1,23 +0,0 @@
...
-
-command /showmotd:
-	description: Displays the message of the day
-	trigger:
-		message {motd}
-
-on join:
-	# uncomment the following line to make the MOTD appear after other messages
-	# wait a tick
-	message {motd}
```
Comment (JRoy) [inline src/main/resources/scripts/simple motd.sk]: Why is this removed?

File: src/main/resources/scripts/realistic drops.sk [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+on break of glass:
+	drop glass
+
+on break of glass pane:
+	drop glass pane
+
+on break of bookshelf:
+	cancel event
+	set block to air
+	drop bookshelf
```
Comment (JRoy) [inline src/main/resources/scripts/realistic drops.sk]: Remove unnecessarily large diff (Also if this remains colored glass support should be added)

File: src/main/resources/scripts/plant with hoe.sk [hunk 4]
```diff
@@ -1,32 +0,0 @@
...
-
-# right click sand to plant a cactus
-on rightclick on sand with any hoe:
-	player has a cactus
-	block above is air
-	blocks north above, east above, south above and west above are allowed next to cactus
-	
-	damage tool by 1
-	set block above to cactus
-	remove a cactus from player
```
Comment (JRoy) [inline src/main/resources/scripts/plant with hoe.sk]: Why was this removed?

File: src/main/resources/scripts/furnace automatisation.sk [hunk 5]
```diff
@@ -1,43 +0,0 @@
...
-			set ore of event-block to loop-item
-			stop trigger
-
-on smelt:
-	loop blocks in radius {@radius}:
-		loop-block is chest
-		loop-block can hold result of event-block
-		add result of event-block to loop-block
-		clear result of event-block
-		stop trigger
```
Comment (JRoy) [inline src/main/resources/scripts/furnace automatisation.sk]: Why was this removed?

File: src/main/resources/scripts/vanilla gui.sk (hunk 1 above)
Comment (Olyno) [reply src/main/resources/scripts/vanilla gui.sk]: > How would the user have this script without skript 2.3 :P

With only Skript? He can't.

The user can't make it run without skript 2.3+, but if this pull request is merged, it should not be a problem 'cause it will be Skript 2.4+

File: src/main/resources/scripts/realistic drops.sk (hunk 3 above)
Comment (Olyno) [reply src/main/resources/scripts/realistic drops.sk]: This large diff is caused by the rename of the file. About colored glass, I can add it of course!

File: src/main/resources/scripts/vanilla gui.sk (hunk 1 above)
Comment (JRoy) [reply src/main/resources/scripts/vanilla gui.sk]: Exactly my point, this will come bundled with skript 2.4, no need to add a skript 2.3 notice

File: src/main/resources/scripts/plant with hoe.sk (hunk 4 above)
Comment (Olyno) [reply src/main/resources/scripts/plant with hoe.sk]: In my opinion, it's a unecessary example, so I removed it.

File: src/main/resources/scripts/furnace automatisation.sk (hunk 5 above)
Comment (Olyno) [reply src/main/resources/scripts/furnace automatisation.sk]: Hmm 2 possiblities: was a fail or just didn't work on latest skript versions. I think it's the 2nd possibility. I should fix it later.

File: src/main/resources/scripts/vanilla gui.sk (hunk 1 above)
Comment (Olyno) [reply src/main/resources/scripts/vanilla gui.sk]: Yeah but I prefer to add this precision if any user want to share the example.

Comment (JRoy) [reply src/main/resources/scripts/vanilla gui.sk]: In that case, it's in french not english :p

File: src/main/resources/scripts/simple motd.sk (hunk 2 above)
Comment (Olyno) [reply src/main/resources/scripts/simple motd.sk]: It is not removed, just remade here: https://github.com/SkriptLang/Skript/pull/2559/files#diff-18e684da942462e3df6b8fc506c3aa90

File: src/main/resources/scripts/vanilla gui.sk (hunk 1 above)
Comment (Olyno) [reply src/main/resources/scripts/vanilla gui.sk]: omg I didn't see this fault, sorry about that, I will fix it.

File: src/main/resources/scripts/realistic drops.sk (hunk 3 above)
Comment (JRoy) [reply src/main/resources/scripts/realistic drops.sk]: you didn't rename the file, it's just a line-ending issue

File: src/main/resources/scripts/simple motd.sk (hunk 2 above)
Comment (JRoy) [reply src/main/resources/scripts/simple motd.sk]: Join messages/Quit messages are different from MOTD's

MOTD's are sent to a player only when they join, join/quit messages are sent to everyone when a player joins/leaves.

Comment (Olyno) [reply src/main/resources/scripts/simple motd.sk]: Is it not a useless example? I mean, in 2 lines you can do it:
```vb
on join:
    send "Welcome back %player%!"
```
Seems useless to add an example for that.

Comment (Blueyescat) [reply src/main/resources/scripts/simple motd.sk]: Those are examples to teach something to the users, not snippets. Anyways it is still an useful system, and the code teaches the Skript user useful things.

File: src/main/resources/scripts/plant with hoe.sk (hunk 4 above)
Comment (Blueyescat) [reply src/main/resources/scripts/plant with hoe.sk]: ~~or you were lazy to fix the code~~

Like i said above, examples are to teach users something.

I think you can use the `is solid` condition instead of the `allowed next to cactus` alias here (needs testing).

**Comment by Olyno** [comment]
> Preferably have the inventory set to a variable, modify it there, then open it to the player. Optionally use metadata for verifying the inventory instead of names.

Can you quote lines about what are you talking please? Should be mroe easily to fix/add this suggestion.

**Comment by bluelhf** [comment]
@Olyno Believe me, I am trying.

**Comment by ShaneBeee** [comment]
> @Olyno Believe me, I am trying.

Go to the code (by clicking `files changed` at the top)
Scroll to the line you think needs to be changed, and a little button should pop up, click it, and you can comment
Screen Shot 2019-10-13 at 3 22 29 PM
:

File: src/main/resources/scripts/vanilla gui.sk [hunk 6]
```diff
@@ -0,0 +1,15 @@
+#
+# A vanilla example of a gui creation using Skript only
+# Requires Skript 2.3+
+#
+
+command /gui:
+	trigger:
+		open chest inventory with 6 row with name "Gui Test" to player
+		set slot 0 of open inventory of player to stone
```
Comment (bluelhf) [inline src/main/resources/scripts/vanilla gui.sk]: After a change of browser and desktop mode, here I am:
Like I said, local variables for the inventory and metadata for checking it.

Comment (Olyno) [reply src/main/resources/scripts/vanilla gui.sk]: Updated

File: src/main/resources/scripts/furnace automatisation.sk [hunk 7]
```diff
@@ -1,43 +1,38 @@
...
+	#   1.5: chests diaginally adjacent (i.e. which touch the furnace with one edge) are affected as well
+	#   1.75: all surrounding chests in a 3x3x3 cube are affected
+	radius: 1
+
+aliases:
+	fuel = coal, coal ore, coal block, any wooden tool
+
+on smelt:
+	loop blocks in radius {@radius}:
+		loop-block is chest
```
Comment (Wealthyturtle) [inline src/main/resources/scripts/furnace automatisation.sk]: Use an if statement here, for consistency's sake

Comment (Wealthyturtle) [reply src/main/resources/scripts/furnace automatisation.sk]: Same thing for the lines below, e.g. ``inventory of loop-block contains loop-item``

File: src/main/resources/scripts/homes.sk [hunk 8]
```diff
@@ -1,22 +1,30 @@
 #
 # A simple home script which allows players with the permission 'skript.home' to
-# define a home location with /sethome and teleport to that location whenever they want to with /home.
+# define a home location with /sethome, delete home with /delhome and teleport to that location whenever they want to with /home.
```
Comment (Wealthyturtle) [inline src/main/resources/scripts/homes.sk]: delete **a** home with /delhome

File: src/main/resources/scripts/homes.sk [hunk 9]
```diff
@@ -1,22 +1,30 @@
...
 	executable by: players
 	trigger:
-		set {homes.%player%.location} to location of player
-		message "Set your home to <grey>%location of player%<reset>"
+		set {homes::%uuid of player%::%arg-1%} to location of player
+		message "Set your home <green>%arg-1%<reset> to <grey>%location of player%<reset>"
 
-command /home:
+command /delhome <string>:
+	description: Delete your home
```
Comment (Wealthyturtle) [inline src/main/resources/scripts/homes.sk]: Delete**s** your home

File: src/main/resources/scripts/homes.sk [hunk 10]
```diff
@@ -1,22 +1,30 @@
 #
 # A simple home script which allows players with the permission 'skript.home' to
-# define a home location with /sethome and teleport to that location whenever they want to with /home.
+# define a home location with /sethome, delete home with /delhome and teleport to that location whenever they want to with /home.
 #
 
-command /sethome:
+command /sethome <string>:
 	description: Set your home
```
Comment (Wealthyturtle) [inline src/main/resources/scripts/homes.sk]: Set**s** your home

File: src/main/resources/scripts/furnace automatisation.sk [hunk 11]
```diff
@@ -1,43 +1,38 @@
...
-	loop blocks in radius {@radius}:
-		loop-block is chest
-		loop-block can hold result of event-block
-		add result of event-block to loop-block
-		clear result of event-block
-		stop trigger
+#
+# This file automates furnaces.
+# They store smelted items, refuel and restock automatically in/from surrounding chests
+# But only as long as they have fuel and something to smelt, and there's storage space for the smolten item(s).,
```
Comment (Wealthyturtle) [inline src/main/resources/scripts/furnace automatisation.sk]: Shouldn't be ``smolten item(s).``, should just be ``smelted items`` (remove the full stop too)

File: src/main/resources/scripts/furnace automatisation.sk [hunk 12]
```diff
@@ -1,43 +1,38 @@
...
+	# radius: in which radius chest should be searched.
+	# Putting a small radius is recommended as it increases performance and prevents furnaces from taking items out of chests behind walls or otherwise hidden chests.
+	# Some recommended values:
+	#   1: only chests direcly next to the furnace are affected
+	#   1.5: chests diaginally adjacent (i.e. which touch the furnace with one edge) are affected as well
+	#   1.75: all surrounding chests in a 3x3x3 cube are affected
+	radius: 1
+
+aliases:
+	fuel = coal, coal ore, coal block, any wooden tool
```
Comment (Wealthyturtle) [inline src/main/resources/scripts/furnace automatisation.sk]: Any wood log? Lava bucket? Also I'm quite sure coal ore shouldn't be a fuel source

File: src/main/resources/scripts/homes.sk [hunk 13]
```diff
@@ -1,22 +1,30 @@
...
+		message "Set your home <green>%arg-1%<reset> to <grey>%location of player%<reset>"
 
-command /home:
+command /delhome <string>:
+	description: Delete your home
+	permission: skript.home
+	executable by: players
+	trigger:
+		clear {homes::%uuid of player%::%arg-1%}
+		message "Delete your home <green>%arg-1%<reset>"
```
Comment (Wealthyturtle) [inline src/main/resources/scripts/homes.sk]: Delete**d** your home

File: src/main/resources/scripts/realistic drops.sk [hunk 14]
```diff
@@ -1,14 +1,14 @@
...
-
-on break of glass pane:
-	drop glass pane
-
-on break of bookshelf:
-	cancel event
-	set block to air
-	drop bookshelf
+#
+# these triggers fix some drops.
```
Comment (Wealthyturtle) [inline src/main/resources/scripts/realistic drops.sk]: **T**hese triggers fix some drops.

File: src/main/resources/scripts/simple join and leave message.sk [hunk 15]
```diff
@@ -0,0 +1,27 @@
+#
+# A join and leave message.
+#
+
+command /set <string> message <string>:
+	permission: skript.set.join message
```
Comment (Wealthyturtle) [inline src/main/resources/scripts/simple join and leave message.sk]: Space between ``skript.set.join`` and ``message`` may confuse new people

File: src/main/resources/scripts/simple motd.sk [hunk 16]
```diff
@@ -1,23 +1,23 @@
...
+#
+# A simple message of the day script.
+# The MOTD can be set with the /setmotd command,
+# and it will be sent to each player when they log in
+# and when they use the /motd command.
+#
+
+command /setmotd <text>:
+	permission: skript.setmotd
+	description: Sets message of the day
```
Comment (Wealthyturtle) [inline src/main/resources/scripts/simple motd.sk]: Sets **the** message of the day

File: src/main/resources/scripts/vanilla gui.sk [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+	trigger:
+		set {_gui} to a new chest inventory with 6 row with name "Gui Test"
+		set slot 0 of {_gui} to stone
+		open {_gui} to player
+ 
+on inventory click:
+	if name of event-inventory is "Gui Test":
+		if index of event-slot = 0:
+			cancel event
+			send "You clicked on the slot 0 !"
```
Comment (Wealthyturtle) [inline src/main/resources/scripts/vanilla gui.sk]: Should be using the message effect, not send, for consistency sake.

File: src/main/resources/scripts/furnace automatisation.sk [hunk 18]
```diff
@@ -1,43 +1,38 @@
...
-
-on smelt:
-	loop blocks in radius {@radius}:
-		loop-block is chest
-		loop-block can hold result of event-block
-		add result of event-block to loop-block
-		clear result of event-block
-		stop trigger
+#
+# This file automates furnaces.
```
Comment (Wealthyturtle) [inline src/main/resources/scripts/furnace automatisation.sk]: This **script** automates furnaces.

File: src/main/resources/scripts/furnace automatisation.sk [hunk 19]
```diff
@@ -1,43 +1,38 @@
...
+# They store smelted items, refuel and restock automatically in/from surrounding chests
```
Comment (Wealthyturtle) [inline src/main/resources/scripts/furnace automatisation.sk]: They store smelted items **and restock/refuel automatically from** surrounding chests

File: src/main/resources/scripts/item command.sk [hunk 20]
```diff
@@ -19,11 +19,11 @@ command /item <items>:
 	executable by: players
 	permission: skript.give
 	trigger:
-		player has permission "skript.give.bypassblacklist":
+		if player has permission "skript.give.bypassblacklist":
 			give arguments to player
 		else:
 			loop arguments:
-				loop-item is not blacklisted:
+				if loop-item is not blacklisted:
```
Comment (Wealthyturtle) [inline src/main/resources/scripts/item command.sk]: Blacklisted is not a condition?

File: src/main/resources/scripts/teleport with compass.sk [hunk 21]
```diff
@@ -3,14 +3,8 @@
 # with a compass like in WorldEdit.
 #
 
-aliases:
-	# blocks without collision as of Minecraft 1.5 (excluding fire, nether portal & end portal)
-	nonsolid = 0, 6, 27, 28, 30, 31, 32, 37, 38, 39, 40, 50, 55, 59, 63, 64, 65, 66, 68, 69, 70, 71, 72, 75, 76, 77, 78:0, 83, 85:4-7, 104, 105, 106, 111, 115, 131, 132, 141, 142, 143, 147, 148, 157
-
 on rightclick with compass:
 	player has permission "skript.teleport"
```
Comment (Wealthyturtle) [inline src/main/resources/scripts/teleport with compass.sk]: Inconsistent conditional

File: src/main/resources/scripts/item command.sk (hunk 20 above)
Comment (TPGamesNL) [reply src/main/resources/scripts/item command.sk]: Look at the top of the file, blacklisted is defined as an alias

Comment (Wealthyturtle) [reply src/main/resources/scripts/item command.sk]: Oh my bad. Should probably add a comment that it’s an alias and not an inbuilt conditional then

File: src/main/resources/scripts/teleport with compass.sk (hunk 21 above)
Comment (Olyno) [reply src/main/resources/scripts/teleport with compass.sk]: I updated the scripts to work with the latest version of Skript. So I kept things to a maximum.

File: src/main/resources/scripts/kill counter.sk [hunk 22]
```diff
@@ -5,22 +5,22 @@
 # Defaults are useful here, as a player's kills would be '<none>' if he hasn't killed anything yet,
 # but with defaults defined those will be used in that case.
 variables:
-	{kill counter.%player%.kills_session} = 0
-	{kill counter.%player%.kills_session_max} = 0
-	{kill counter.%player%.kills_total} = 0
+	{kill counter::%player%::kills_session} = 0
```
Comment (bluelhf) [inline src/main/resources/scripts/kill counter.sk]: Player names in variables is bad practice. Preferrably use UUIDs.

File: src/main/resources/scripts/vanilla gui.sk (hunk 17 above)
Comment (Olyno) [reply src/main/resources/scripts/vanilla gui.sk]: Both are the same. We should let 2 different things to show to any user that oth possibilities work.

File: src/main/resources/scripts/kill counter.sk (hunk 22 above)
Comment (Olyno) [reply src/main/resources/scripts/kill counter.sk]: Can't here, that's the reason why I didn't use it here. Variables doesn't work with ``uuid of player``.

File: src/main/resources/scripts/furnace automatisation.sk (hunk 7 above)
Comment (Olyno) [reply src/main/resources/scripts/furnace automatisation.sk]: Again, it's an ``affirmative`` condition, so it shoud not contain any ``if`` before.

File: src/main/resources/scripts/kill counter.sk (hunk 22 above)
Comment (bluelhf) [reply src/main/resources/scripts/kill counter.sk]: They do. Variables don't work with the uuid of a player **variable**.

Comment (bluelhf) [reply src/main/resources/scripts/kill counter.sk]: After testing, confirmed that `Variables` doesn't work with uuid of player, but in that case, the usage of the Default Value expression would be preferred.

Comment (TPGamesNL) [reply src/main/resources/scripts/kill counter.sk]: I think you can just use %string%, since the UUID is a string in Skript.

File: src/main/resources/scripts/teleport with compass.sk (hunk 21 above)
Comment (Wealthyturtle) [reply src/main/resources/scripts/teleport with compass.sk]: Well, since you're updating it you might as well improve on it :P

File: src/main/resources/scripts/vanilla gui.sk (hunk 17 above)
Comment (Wealthyturtle) [reply src/main/resources/scripts/vanilla gui.sk]: New skript users may not know the difference, if you want to use send, would be nice to add a comment that the send effect is identical to the message effect

Comment (Olyno) [reply src/main/resources/scripts/vanilla gui.sk]: I think not it will be useful to add that

**Comment by Matocolotoe** [comment]
An UUID lol

File: src/main/resources/scripts/realistic drops.sk (hunk 3 above)
Comment (Olyno) [reply src/main/resources/scripts/realistic drops.sk]: Outdated issue + don't know how resolve a commit with a line-ending issue without add any ``.gitattribute`` file

**Comment by Nicofisi** [comment]
Is it now okay to merge this? Everything of importance seems to have been resolved, and this is better than what we present to new users now, looks like quality work.

*People could pull request additional suggestions separately, because otherwise this is never getting merged...*

**Comment by Olyno** [comment]
> Is it now okay to merge this? Everything of importance seems to have been resolved, and this is better than what we present to new users now, looks like quality work.
> 
> _People could pull request additional suggestions separately, because otherwise this is never getting merged..._

I didn't have the time to readd and fix ``plant with hoe.sk``, except that, everything should be ok in this pull request
